src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -210,8 +217,8 @@ ZoneSet::ZonesFromPoint(POINT pt) const noexcept
     {
         for (size_t j = i + 1; j < capturedZones.size(); ++j)
         {
-            const auto& rectI = m_zones[capturedZones[i]]->GetZoneRect();
-            const auto& rectJ = m_zones[capturedZones[j]]->GetZoneRect();
+            const auto& rectI = m_zones.at(capturedZones[i])->GetZoneRect();
```

Comment:
same as above in multiple places regarding `const auto&`.

also, we're now using `.at` instead of `.operator[]` everywhere, which throws instead of emplacing a new element when trying to access non-existing element. do we need to handle `std::out_of_range` then?

Reply:
Added try-catch. Thanks